src/createAnimatedComponent.tsx

```diff
@@ -412,32 +422,51 @@ export default function createAnimatedComponent(
         // update UI props whitelist for this view
         if (
           hostInstance &&
-          this._hasReanimated2Props(styles) &&
+          (this.props.animatedProps?.viewDescriptors || styles.length) &&
           hostInstance.viewConfig
         ) {
           adaptViewConfig(hostInstance.viewConfig);
         }
       }
       this._viewTag = viewTag as number;
 
-      styles.forEach((style) => {
-        if (style?.viewDescriptors) {
-          style.viewDescriptors.add({ tag: viewTag, name: viewName });
-          if (isJest()) {
-            /**
-             * We need to connect Jest's TestObject instance whose contains just props object
-             * with the updateProps() function where we update the properties of the component.
-             * We can't update props object directly because TestObject contains a copy of props - look at render function:
-             * const props = this._filterNonAnimatedProps(this.props);
-             */
-            this.animatedStyle.value = {
-              ...this.animatedStyle.value,
-              ...style.initial.value,
-            };
-            style.animatedStyle.current = this.animatedStyle;
+      // remove old styles
+      if (prevStyles) {
+        const hasOneSameStyle =
+          styles.length === 1 &&
+          prevStyles.length === 1 &&
+          styles[0] === prevStyles[0]; // optimization
```

Comment:
document that in most of the cases views have only a single animated style and it remains unchanged (it is what useAnimatedStyle returns)

Also it'd be worth checking if we can actually use equality checks to compare useAnimatedStyle outputs

Reply:
> Also it'd be worth checking if we can actually use equality checks to compare useAnimatedStyle outputs

Indeed, equality check didn't work as expected. Fixed in 21f456f1e41d5a94b82e95376fb26957864c36e7. For now, comparing `style.viewsRef` should be enough, as this prop is assigned only once (so the reference is the same). In the future, we may introduce some auto-incremented ID field to `useAnimatedStyle` output.